src/libraries/Microsoft.PowerFx.Connectors/ODataQueryableTableValue.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Specialized;
+using System.Globalization;
+using System.Web;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Interpreter;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public abstract class ODataQueryableTableValue : QueryableTableValue
+    {
+        private readonly Uri _uriBase;
+        private readonly ODataParams _odataParams;
+
+        protected ODataQueryableTableValue(TableType tableType, Uri uriBase, ODataParams odataParams = default)
+            : base(IRContext.NotInSource(tableType))
+        {
+            _uriBase = uriBase;
+            _odataParams = odataParams;
+        }
+
+        protected abstract ODataQueryableTableValue WithParameters(ODataParams odataParamsNew);
+
+        public Uri GetUri()
```

Comment:
@MikeStall do you still want me to remove _uriBase and instead have it as a parameter here to GetUri? or maybe even replace this method with `NameValueCollection GetQueryParameters()`?

Reply:
I'd remove _uriBase from here and keep in the derive class.

What about just expose ODataParams as a property - it's already a bublic type and a key field on this class - so is there a benefit to not expose? 

From there, the other methods can be helpers on that ODataParams:

1. AddTo(NameValueCollection query) is already present. 
2. ODataParams.GetUri(uriBase) - this is a convenience method.

Reply:
exposed `ODataParams` and added convenience `GetUri(uriBase)`. Removed `_uriBase` from base class